blocks/shared-styles/_children/promo-image/index.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { useFusionContext } from 'fusion:context';
+import { useContent } from 'fusion:content';
+import getProperties from 'fusion:properties';
+import { extractImageFromStory, extractResizedParams, ratiosFor } from '@wpmedia/resizer-image-block';
+import { Image } from '@wpmedia/engine-theme-sdk';
+import PlaceholderImage from '@wpmedia/placeholder-image-block';
+import { PromoLabel } from '@wpmedia/shared-styles';
+import discoverPromoType from './discover';
+
+import './styles.scss';
+
+const PromoImage = ({
+  content = {},
+  customImageURL = null,
+  alt = '',
+  linkURL,
+  newTab,
+  promoSize,
+  imageRatio,
+  showPromoLabel = false,
+  promoLabelSize = 'small',
+  lazyLoad = false,
+}) => {
+  const { arcSite, isAdmin } = useFusionContext();
+  const promoType = showPromoLabel ? discoverPromoType(content) : null;
+  const ratios = ratiosFor(promoSize, imageRatio);
+
+  let imageConfig = null;
+  if (
+    (customImageURL && lazyLoad)
+    || (customImageURL && isAdmin)) {
+    imageConfig = 'resize-image-api-client';
+  } else if (customImageURL) {
+    imageConfig = 'resize-image-api';
+  }
+
+  const customFieldImageResizedImageOptions = useContent({
```

Comment:
wait, I thought this didn't work as expected? was this supposed to be a class component?